schema.json

```diff
@@ -101,58 +103,51 @@
             "type": "string",
             "format": "regex"
         },
-        "additionalItems": { "$ref": "#" },
+        "additionalItems": { "$recursiveRef": "#" },
         "items": {
             "anyOf": [
-                { "$ref": "#" },
+                { "$recursiveRef": "#" },
                 { "$ref": "#/$defs/schemaArray" }
-            ]
+            ],
+            "default": true
         },
         "maxItems": { "$ref": "#/$defs/nonNegativeInteger" },
         "minItems": { "$ref": "#/$defs/nonNegativeIntegerDefault0" },
         "uniqueItems": {
             "type": "boolean",
             "default": false
         },
-        "contains": { "$ref": "#" },
+        "contains": { "$recursiveRef": "#" },
         "maxProperties": { "$ref": "#/$defs/nonNegativeInteger" },
         "minProperties": { "$ref": "#/$defs/nonNegativeIntegerDefault0" },
         "required": { "$ref": "#/$defs/stringArray" },
-        "additionalProperties": { "$ref": "#" },
+        "additionalProperties": { "$recursiveRef": "#" },
         "properties": {
             "type": "object",
-            "additionalProperties": { "$ref": "#" },
+            "additionalProperties": { "$recursiveRef": "#" },
             "default": {}
         },
         "patternProperties": {
             "type": "object",
-            "additionalProperties": { "$ref": "#" },
+            "additionalProperties": { "$recursiveRef": "#" },
             "propertyNames": { "format": "regex" },
             "default": {}
         },
-        "dependentSchemas": {
```

Comment:
I rather like this being explicitly defined as a schema dependency vs a property dependency.  Why are you changing back to just `dependencies`?

Reply:
That was unintentional and I have no idea, thanks for catching.  I blame coming back to this after three months and probably my tree was in a weird state.